client/src/components/Popup.tsx

```diff
@@ -19,6 +20,23 @@ interface PopupProps {
  * Clicking on the backdrop will close the popup.
  */
 const Popup = (props: PopupProps) => {
+    const { enabled, setEnabled, children, className } = props;
```

Comment:
Why did we deconstruct the props? We can just use `props.enabled` instead of doing all this -- you have created unneeded work and extra unused variables. See below where we use `props.className` and `props.children`.